diff --git a/node_modules/@revenuecat/purchases-capacitor-ui/ios/Sources/RevenuecatPurchasesCapacitorUI/RevenueCatUIPlugin.swift b/node_modules/@revenuecat/purchases-capacitor-ui/ios/Sources/RevenuecatPurchasesCapacitorUI/RevenueCatUIPlugin.swift
index ee8b672..e97cc0b 100644
--- a/node_modules/@revenuecat/purchases-capacitor-ui/ios/Sources/RevenuecatPurchasesCapacitorUI/RevenueCatUIPlugin.swift
+++ b/node_modules/@revenuecat/purchases-capacitor-ui/ios/Sources/RevenuecatPurchasesCapacitorUI/RevenueCatUIPlugin.swift
@@ -192,7 +192,7 @@ public class RevenueCatUIPlugin: CAPPlugin, CAPBridgedPlugin {
     private func resolvePurchaseLogicResult(_ call: CAPPluginCall) {
         call.withRequiredRequestId { requestId in
             let resultStr = call.getString("result") ?? HybridPurchaseLogicBridge.resultError
-            let errorMessage = (call.getObject("error") as? [String: Any])?["message"] as? String
+            let errorMessage = call.getObject("error")?["message"] as? String
             if #available(iOS 15.0, *) {
                 HybridPurchaseLogicBridge.resolveResult(
                     requestId: requestId,
@@ -328,7 +328,7 @@ private extension RevenueCatUIPlugin {
         var options: [String: Any] = [:]
         if let offeringIdentifier = offeringIdentifier {
             options[PaywallProxy.PaywallOptionsKeys.offeringIdentifier] = offeringIdentifier
-            if let presentedOfferingContext = presentedOfferingContext,
+            if presentedOfferingContext != nil,
                 let contextOfferingIdentifier = contextOfferingIdentifier {
                 var presentedOfferingContextMap = [
                     PaywallProxy.PresentedOfferingContextKeys.offeringIdentifier: contextOfferingIdentifier,
